rfcs/0050-vm-syscalls-3/0050-vm-syscalls-3.md

```diff
@@ -0,0 +1,217 @@
+---
+Number: "0050"
+Category: Standards Track
+Status: Draft
+Author: Xu Jiandong <lynndon@gmail.com>, Dingwei Zhang <zhangsoledad@gmail.com>
+Created: 2023-04-17
+---
+
+# VM Syscalls 3
+
+## Abstract
+
+This document describes the addition of the syscalls during the CKB2023. This update significantly enhances the flexibility of CKB Script.
+
+The following four syscalls are added:
+
+- [Spawn]
+- [Get Memory Limit]
+- [Current Memory]
+- [Set Content]
+- [Load Block Extension]
+
+### Spawn
+[Spawn]: #spawn
+
+The syscall Spawn is the core part of this update. The *Spawn* and the latter two syscalls: *Get Memory Limit* and *Set Content* together, implement a way to call another CKB Script in a CKB Script. Unlike the *Exec*[1](../0034-vm-syscalls-2/0034-vm-syscalls-2.md) syscall, *Spawn* saves the execution context of the current script, like [posix_spawn](https://man7.org/linux/man-pages/man3/posix_spawn.3.html), the parent script blocks until the child script ends.
```

Comment:
I think it's better to emphasize that the parent script continues instead of that it blocks.
```suggestion
The syscall Spawn is the core part of this update. The *Spawn* and the latter two syscalls: *Get Memory Limit* and *Set Content* together, implement a way to call another CKB Script in a CKB Script. Unlike the *Exec*[1](../0034-vm-syscalls-2/0034-vm-syscalls-2.md) syscall, *Spawn* saves the execution context of the current script, like [posix_spawn](https://man7.org/linux/man-pages/man3/posix_spawn.3.html), the parent script continues when the child script ends.
```